lib/msf/core/exploit/remote/kerberos/service_authenticator/base.rb

```diff
@@ -183,6 +184,23 @@ def rport
     port
   end
 
+  def connect(options = {})
```

Comment:
Placeholder comment: For something like a kerberos smb login bruteforcer, will this end making hundreds of DNS requests

Reply:
`CachedResolver` doesn't care how many requests you make, so long as you're within the record's TTL (or create a static entry after the first lookup in the brute mixin) it'll pull from the cache.

Reply:
Yeah I was thinking that caching should be the resolvers responsibility if it needs to be implemented.

Reply:
https://github.com/rapid7/metasploit-framework/blob/master/lib/rex/proto/dns/cached_resolver.rb#L13 :)